### PR TITLE
feat(curriculum): add book catalog table lab

### PR DESCRIPTION
**Pull request by moT01**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

I changed up the user stories quite a bit to be a little less hand holding. [Here are the originals](https://github.com/freeCodeCamp/CurriculumExpansion/blob/main/frontend-cert/html-projects/book-catalogue-table/user-stories.md)

### CODE REVIEW

**Comment by jdwilkin4** [comment]
Is this ready for review or still in draft mode?

File: curriculum/challenges/english/25-front-end-development/lab-book-catalog-table/66ec4c8e9878d8441956516f.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+const numberOfBooks = document.querySelectorAll('tbody tr')?.length;
+assert.equal(document.querySelector('tfoot tr td').textContent, `Total Books: ${numberOfBooks}`);
+```
+
+# --seed--
+
+## --seed-contents--
+
+```html
+
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-book-catalog-table/66ec4c8e9878d8441956516f.md:124]:
```suggestion
<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Book Catalog</title>
</head>

<body>
</body>

</html>
```

File: curriculum/challenges/english/25-front-end-development/lab-book-catalog-table/66ec4c8e9878d8441956516f.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+You should have one `thead` element within your `table` element.
+
+```js
+assert.lengthOf(document.querySelectorAll('table thead'), 1);
+```
+
+You should have one `tr` element within your `thead` element.
+
+```js
+assert.lengthOf(document.querySelectorAll('table thead'), 1);
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-book-catalog-table/66ec4c8e9878d8441956516f.md:40]:
```suggestion
assert.lengthOf(document.querySelectorAll('thead tr'), 1);
```

File: curriculum/challenges/english/25-front-end-development/lab-book-catalog-table/66ec4c8e9878d8441956516f.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+});
+```
+
+You should have one `tfoot` element within your `table` element.
+
+```js
+assert.lengthOf(document.querySelectorAll('table tfoot'), 1);
+```
+
+You should have exactly one `tr` elemenet in your `tfoot` element.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-book-catalog-table/66ec4c8e9878d8441956516f.md:100]:
```suggestion
You should have exactly one `tr` element in your `tfoot` element.
```

**Review by Dario-DC — COMMENTED** [review]
Left a couple of comments. Apart from that it looks good :+1:

**Review by Dario-DC — APPROVED** [review]

**Review by zairahira — APPROVED** [review]

**Pull request merged**